### ISSUE TITLE
No loader for `editor/assets/stylesheets/main.scss`

### ISSUE THREAD

**Issue by andreiglingeanu** [WordPress/gutenberg#3129]
## Issue Overview

The file `editor/assets/stylesheets/main.scss` does not get covered by any webpack loader and because of that, the bundle does not get properly compiled.

## Steps to Reproduce (for bugs)

1. Clone the laster `master` branch of gutenberg: `git clone git@github.com:WordPress/gutenberg.git`
2. run `npm install`
3. run `npm run build` (`npm run cross-env BABEL_ENV=default webpack` to run into dev environment)

## Expected Behavior

Expected the whole webpack bundle to get compiled with success.

## Current Behavior

```
ERROR in ./editor/assets/stylesheets/main.scss
Module parse failed: /Users/andreiglingeanu/Projects/full/wp_installs/ctguttenbergblocks/wp-content/plugins/gutenberg/editor/assets/stylesheets/main.scss Unexpected token (1:27)
You may need an appropriate loader to handle this file type.
| body.gutenberg-editor-page {
|       background: $white;
|
 @ ./editor/index.js 16:0-40
```

Also, I get an error in PHP from `lib/client-assets.php` file but that's probably a consequence of the webpack bundle not getting properly compiled down.

## Possible Solution

As far as I observed, the file `./editor/assets/stylesheets/main.scss` is not covered by rule in webpack config. I have tried to add manually an entry into the `module.rules` webpack section with `sass-loader` but that didn't worked properly for me.

## Related Issues and/or PRs

Haven't seen any of them. Have asked in Slack #core-editor but without any success.

## Software

```
Node version: v8.4.0
NPM version: 5.3.0
```

## Todos
- [ ] Tests
- [ ] Fix webpack config

**Comment by aduth** [comment]
`main.scss` is covered by this Webpack loader:

https://github.com/WordPress/gutenberg/blob/1b68a192410eaeba5e494f6f4334045b0bc65a5a/webpack.config.js#L130-L137

Following your instructions from a fresh clone, the build works fine for me.

Can you share your version of Node / npm ? (`node --version`, `npm --version`)

**Comment by andreiglingeanu** [comment]
Hey @aduth, just added the software versions into the issue body. Sorry for not adding that in the first place.

**Comment by aduth** [comment]
I am running similar versions, except a newer npm version. You can try upgrading (`npm i -g npm`) and installing dependencies again (`rm -rf node_modules && npm i`) but I have a feeling this will not resolve any issue.

You are certain this issue occurs when cloned fresh, no local changes (`git status` clean) and on master branch?

**Comment by andreiglingeanu** [comment]
Yes, I was indeed having a clean staging area.

But, I just cloned a fresh copy of gutenberg in `/tmp` and installed the dependencies and run the `npm run build` script from there. Worked fine.

When doing the described actions in a folder somewhere under my `~/projects` it fails. Probably because of the way webpack's module lookup mechanism is picking up modules.

**Comment by aduth** [comment]
>Probably because of the way webpack's module lookup mechanism is picking up modules.

Sounds like that could be a bug in Webpack. Maybe one which would be fixed by us upgrading to the latest version?

**Comment by andreiglingeanu** [comment]
I would love to try my hand at it.

**Comment by andreiglingeanu** [comment]
The latest [`3.8.1`](https://github.com/webpack/webpack/releases/latest) does not actually fix that issue, just tried it.

**Comment by andreiglingeanu** [comment]
@aduth I addressed the webpack upgrade in #3130.
